src/components/FiltersBox.test.js

```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+import {
+  render, cleanup, waitForElementToBeRemoved, fireEvent, wait,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { Simulate } from 'react-dom/test-utils';
+
+import Provider from '../contexts/Provider';
+import Table from './Table';
+import { SW_BASE_API } from '../services/SWapi';
+import mockFetchPromise, { planets } from '../__mocks__/apiMock';
+
+const testTableRender = async () => {
+  jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);
+
+  const { queryByText, getByRole, queryByTestId } = render(
```

Comment:
Tente trabalhar com uma função que renderiza as queries, vai te poupar umas boas linhas de código.

Exportando ela e trabalhando com o recebimento do componente no parâmetro, seria mais prático...

```suggestion
  const { queryByText, getByRole, queryByTestId } = renderWithRouter(<Table />)
```

Dê uma olhada em renderWithRouter.

Reply:
Muito legal sua observação, eu nao usei o renderWithRouter pq n to trabalhando com nenhum tipo de rota, mas vou dar uma olhada se tem algo parecido com ele pra trabalhos que nao utilizam rotas.